pandas/_libs/tslibs/resolution.pyx

```diff
@@ -0,0 +1,649 @@
+# -*- coding: utf-8 -*-
+# cython: profile=False
+
+from cython cimport Py_ssize_t
+
+import numpy as np
+cimport numpy as np
+from numpy cimport ndarray, int64_t
+np.import_array()
+
+from util cimport is_string_object, get_nat
+
+from khash cimport (
+    khiter_t,
+    kh_destroy_int64, kh_put_int64,
+    kh_init_int64, kh_int64_t,
+    kh_resize_int64, kh_get_int64)
+
+from cpython.datetime cimport datetime
+
+from np_datetime cimport (pandas_datetimestruct,
+                          dtstruct_to_dt64, dt64_to_dtstruct)
+from frequencies cimport get_freq_code
+from timezones cimport (
+    is_utc, is_tzlocal,
+    maybe_get_tz, get_dst_info, get_utcoffset)
+from fields import build_field_sarray
+from conversion import tz_convert
+
+from pandas._libs.properties import cache_readonly
+from pandas._libs.tslib import Timestamp
+
+from pandas.core.algorithms import unique  # TODO: Avoid this non-cython import
+
+# ----------------------------------------------------------------------
+# Constants
+
+cdef int64_t NPY_NAT = get_nat()
+
+cdef int RESO_NS = 0
+cdef int RESO_US = 1
+cdef int RESO_MS = 2
+cdef int RESO_SEC = 3
+cdef int RESO_MIN = 4
+cdef int RESO_HR = 5
+cdef int RESO_DAY = 6
+
+_ONE_MICRO = 1000L
```

Comment:
can be de-privatized (add to list or do here ok)

Reply:
to the list it goes